Cython/Build/Cythonize.py

```diff
@@ -148,56 +119,93 @@ def run_distutils(args):
                 shutil.rmtree(temp_dir)
 
 
-def parse_args(args):
-    from optparse import OptionParser
-    parser = OptionParser(usage='%prog [options] [sources and packages]+')
+def create_args_parser():
+    from argparse import ArgumentParser, Action
+
+    class ParseDirectivesAction(Action):
+        def __call__(self, parser, namespace, values, option_string=None):
+            old_directives = dict(getattr(namespace, self.dest,
+                                          Options.get_directive_defaults()))
+            directives = Options.parse_directive_list(
+                values, relaxed_bool=True, current_settings=old_directives)
+            setattr(namespace, self.dest, directives)
+
+    class ParseOptionsAction(Action):
+        def __call__(self, parser, namespace, values, option_string=None):
```

Comment:
Maybe make use of the option `type` here? That could solve the `buffer_max_ndims` problem.

Reply:
I would like to keep this PR without behavior changes (also because I think it could become a bigger thing as on might expect, see also https://github.com/cython/cython/pull/2952#discussion_r285398870). So not sure about this.